src/systems/camera-mirror.js

```diff
@@ -0,0 +1,90 @@
+import { paths } from "./userinput/paths";
+
+AFRAME.registerSystem("camera-mirror", {
+  tick() {
+    const userinput = this.el.systems.userinput;
+
+    if (userinput.get(paths.actions.camera.exitMirror) && this.mirrorEl) {
+      this.unmirrorCameraAtEl(this.mirrorEl);
+    }
+  },
+
+  // Adds a camera under el, and starts mirroring
+  mirrorCameraAtEl(el) {
+    // TODO probably should explicitly check for immersive mode here.
+    if (AFRAME.utils.device.isMobile()) return;
+    if (this.mirrorEl && this.mirrorEl !== el) this.unmirrorCameraAtEl(this.mirrorEl);
+
+    this.mirrorEl = el;
+    this.mirrorCamera = new THREE.PerspectiveCamera(80, window.innerWidth / window.innerHeight, 0.1, 30000);
+    this.mirrorCamera.rotation.set(0, Math.PI, 0);
+    el.setObject3D("mirror-camera", this.mirrorCamera);
+
+    const tempHeadScale = new THREE.Vector3();
+    const tempHudScale = new THREE.Vector3();
+
+    const renderer = this.el.renderer;
+
+    if (!this.directRenderFunc) {
+      this.directRenderFunc = renderer.render;
+    }
+
+    const headEl = document.getElementById("player-head");
+    const hudEl = document.getElementById("player-hud");
+    const playerHead = headEl && headEl.object3D;
+    const playerHud = hudEl && hudEl.object3D;
+    document.body.classList.add("mirrored-camera");
+
+    this.el.sceneEl.renderer.render = (scene, camera, renderTarget) => {
+      const wasVREnabled = renderer.vr.enabled;
+
+      if (wasVREnabled) {
+        this.directRenderFunc.call(renderer, scene, camera, renderTarget);
```

Comment:
this isn't really DRY, but it wasn't enough common code for me to feel like breaking it up would be worth the additional complexity of having to parameterize it